update dbt-databricks docs (#3946)
fishtown-analytics/docs.getdbt.com: website/docs/docs/core/connect-data-platform/databricks-setup.md

`dbt-databricks` includes features not available in `dbt-spark`:
- Unity Catalog support
- No need to install additional drivers or dependencies for use on the CLI
- Use of Delta Lake for all models out of the box
- SQL macros that are optimized to run with [Photon](https://docs.databricks.com/runtime/photon.html)

## Connecting to Databricks

To connect to a data platform with dbt Core, create the appropriate _profile_ and _target_ YAML keys/values in the `profiles.yml` configuration file for your Databricks SQL Warehouse/cluster. This dbt YAML file lives in the  `.dbt/` directory of your user/home directory. For more information, refer to [Connection profiles](/docs/core/connect-data-platform/connection-profiles) and [profiles.yml](/docs/core/connect-data-platform/profiles.yml).

`dbt-databricks` can connect to Databricks SQL Warehouses and all-purpose clusters. Databricks SQL Warehouses is the recommended way to get started with Databricks.

Refer to the [Databricks docs](https://docs.databricks.com/dev-tools/dbt.html#) for more info on how
to obtain the credentials for configuring your profile.

## Host parameters

The following profile fields are always required. 

> | Field     | Example | Description |

Reviewer: @ericbaumann not sure if it's just me but i wonder if it would be easier to have the columns set as 'field', 'description', and 'example'. I'm finding myself scanning to the description first before example so i can understand the 'thing'. happy to change this up for you if you agree

Author: that's a great idea! Feel like that will decrease the cognitive load we put on the reader. Thank you @mirnawong1 !!

| Profile field                 | Example                          | Description                                                                                                 |
| ----------------------------- | -------------------------------- | ----------------------------------------------------------------------------------------------------------- |
| `threads`                     | `8`                              | How many threads dbt should use (default is `1`)                                                            |
| `connect_retries`                     | `3`                              | How many times dbt should retry the connection to Databricks (default is `1`)                                                            |
| `connect_timeout`                     | `1000`                              | How many seconds before the connection to Databricks should timeout (default behavior is no timeouts)                                                            |
| `session_properties`          | `ansi_mode: true`         | Sets Databricks session properties used in the connection. Execute `SET -v` to see available options       |

> #### Example profiles.yml for token-based authentication

Reviewer: i'm wondering if this should be an h3 or h2 header? i also like the idea of using tabs so the user can switch between the two without much scrolling. if that makes sense, I'm happy to change this into tabs and share the preview.

Author: Not sure if this is what you're referring to or not, but I was kinda questioning the side nav hierarchy after I saw the deploy preview. I feel like these should maybe be represented on there in some capacity (even if it's one link to the section for `example profiles` - totally defer to your judgment though.

I was also kind of wondering if it'd be appropriate to nest the '[...] Parameters' sections _underneath_ the 'Configuring dbt-databricks' section. What do you think?

Reviewer: yea great shout! i'm going to make some edits if you're ok with that? and will share the deploy

Author: sounds great, thanks mirna!!

- Use of Delta Lake for all models out of the box
- SQL macros that are optimized to run with [Photon](https://docs.databricks.com/runtime/photon.html)

> ### Connecting to Databricks

Reviewer: i would keep this as an h2 for consistency

```suggestion
## Connecting to Databricks
```

> To connect to a data platform with dbt Core, create appropriate _profile_ and _target_ YAML keys/values in the `profiles.yml` configuration file for your Starburst/Trino clusters. This dbt YAML file lives in the  `.dbt/` directory of your user/home directory. For more information, refer to [Connection profiles](/docs/core/connect-data-platform/connection-profiles) and [profiles.yml](/docs/core/connect-data-platform/profiles.yml).

Reviewer: @ericbaumann this references starburst and trino clusters -- is that correct?

```suggestion
To connect to a data platform with dbt Core, create the appropriate _profile_ and _target_ YAML keys/values in the `profiles.yml` configuration file for your Starburst/Trino clusters. This dbt YAML file lives in the  `.dbt/` directory of your user/home directory. For more information, refer to [Connection profiles](/docs/core/connect-data-platform/connection-profiles) and [profiles.yml](/docs/core/connect-data-platform/profiles.yml).
```

Author: Nope! good catch 👀

`dbt-databricks` can connect to Databricks SQL Warehouses and all-purpose clusters. Databricks SQL Warehouses is the recommended way to get started with Databricks.

> See the [Databricks documentation](https://docs.databricks.com/dev-tools/dbt.html#) on how

Reviewer: ```suggestion
Refer to the [Databricks docs](https://docs.databricks.com/dev-tools/dbt.html#) for more info on how
```
